### ISSUE TITLE
track found too late (timeout) on raspberry

### ISSUE THREAD

**Issue by gras64** [forslund/mpd_skill#17]
Good skil. if the database is large, the answer from mpd always comes after the mycroft error message on my raspberry. on PC OK.
the answer comes about 20 seconds after playback control. Do you have an idea how I can speed it up?

` 16:19:13.610 | INFO     |   810 | Playback Control Skill | Resolving Player for: rammstein
 16:19:13.663 | INFO     |   810 | Morse | save output for morse
ft_skill.mycroft_skill:find_resource:720 | Resource 'PlayTheNews.list' for lang 'de-de' not found: trying 'en-us'
Removing event mycroft-playback-control.mycroftai:PlayQueryTimeout
Removing event mycroft-playback-control.mycroftai:PlayQueryTimeout
Removing event mycroft-playback-control.mycroftai:PlayQueryTimeout
 16:19:20.125 | INFO     |   810 | Playback Control Skill |    No matches  >>>>>>>>>
 16:19:20.296 | INFO     |   810 | Morse | save output for morse
Removing event mycroft-mark-2.mycroftai:IdleCheck
 16:19:31.336 | INFO     |   810 | Mark2 | Displaying all animations for page
 16:19:31.338 | INFO     |   810 | Mark2 | Cancelling Idle screen
Removing event mycroft-mark-2.mycroftai:IdleCheck
 16:19:40.228 | INFO     |   810 | MPDSkill | MPD Found Rammstein >>>>>>>`

**Comment by forslund** [comment]
Yes that is an issue at the moment. it's the fuzzy matching that takes time.

There are a couple of ways we can try:
- Don't include tracks in first pass (should make albums and artist work at least)
- See if there's a faster fuzzy matcher maybe [rapidfuzz](https://github.com/maxbachmann/rapidfuzz)
- try to do the fuzzy matching in parallel somehow (don't think python is great for this option though)

**Comment by forslund** [comment]
Gonna keep working on this but I made PR #18 using rapidfuzz. If you get the chance try switching to the branch `feature/rapidfuzz` and reinstall the requirements.txt `mycroft-pip install /opt/mycroft/skills/mpd_skill.forslund/requirements.txt`
